fluentrequest_test.go

```diff
@@ -3,105 +3,154 @@ package fluentrequest
+				Completed: false,
+			},
+			want: ResponseResult{
+				statusCode: http.StatusOK,
+				responseBody: Body{
+					Id:        1,
+					UserId:    1,
+					Title:     "foo",
+					Body:      "bar",
+					Completed: false,
+				},
+			},
+		},
+		{
+			name:   "Test PATCH request",
+			method: http.MethodPatch,
+			url:    "https://jsonplaceholder.typicode.com/todos/1",
+			body: Body{
+				Id:        1,
+				UserId:    1,
+				Title:     "foobar",
+				Body:      "",
+				Completed: true,
+			},
+			want: ResponseResult{
+				statusCode: http.StatusOK,
+				responseBody: Body{
+					Id:        1,
+					UserId:    1,
+					Title:     "foobar",
+					Body:      "",
+					Completed: true,
+				},
+			},
+		},
+		{
+			name:   "Test DELETE request",
+			method: http.MethodDelete,
+			url:    "https://jsonplaceholder.typicode.com/todos/1",
+			want: ResponseResult{
+				statusCode: http.StatusOK,
+			},
+		},
 	}
 
-	resp, err := FluentRequest().
-		Method(http.MethodPost).
-		Url("https://jsonplaceholder.typicode.com/posts").
-		Body(bytes.NewBuffer(data)).
-		Header(header).
-		Run()
-
-	body, _ := ioutil.ReadAll(resp.Body)
-
-	assert.NoError(t, err)
-	assert.Equal(t, expectedPOSTBody, string(body))
-	assert.Equal(t, http.StatusCreated, resp.StatusCode)
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			createRequest(t, tt.method, tt.url, tt.body, tt.want)
+		})
+	}
 }
 
-func TestPUT(t *testing.T) {
-	data, _ := json.Marshal(map[string]interface{}{
-		"id":     1,
-		"title":  "delectus aut autem",
-		"body":   "bar",
-		"userId": 1,
-	})
+func createRequest(t *testing.T, method string, url string, requestBody interface{}, want ResponseResult) {
+	bodyBytes, _ := json.Marshal(requestBody)
 
 	header := http.Header{
 		"Content-Type": {"application/json; charset=UTF-8"},
 	}
 
 	resp, err := FluentRequest().
-		Method(http.MethodPut).
-		Url("https://jsonplaceholder.typicode.com/posts/1").
-		Body(bytes.NewBuffer(data)).
+		Method(method).
+		Body(bytes.NewBuffer(bodyBytes)).
 		Header(header).
+		Url(url).
 		Run()
 
-	body, _ := ioutil.ReadAll(resp.Body)
-
-	assert.NoError(t, err)
-	assert.Equal(t, expectedPUTBody, string(body))
-	assert.Equal(t, http.StatusOK, resp.StatusCode)
-}
-
-func TestPATCH(t *testing.T) {
-	data, _ := json.Marshal(map[string]interface{}{
-		"title": "foo",
-		"body":  "bar",
-	})
+	responseBody, _ := io.ReadAll(resp.Body)
 
-	header := http.Header{
-		"Content-Type": {"application/json; charset=UTF-8"},
-	}
-
-	resp, err := FluentRequest().
-		Method(http.MethodPatch).
-		Url("https://jsonplaceholder.typicode.com/posts/1").
-		Body(bytes.NewBuffer(data)).
-		Header(header).
-		Run()
+	var deserializedBody Body
 
-	body, _ := ioutil.ReadAll(resp.Body)
+	json.Unmarshal(responseBody, &deserializedBody)
 
 	assert.NoError(t, err)
-	assert.Equal(t, expectedPATCHBody, string(body))
-	assert.Equal(t, http.StatusOK, resp.StatusCode)
+	assert.Equal(t, want.responseBody, deserializedBody)
```

Comment:
I was doing some experiments trying to turn our tests more maintenable and just found out the 
assert.JSONEq what is able to assert jsons by strings. 

Going into the code i just saw it does the same Unmarshal as you are doing. 

Using it we can build our Test object with strings in a very simplier and more legible way and removing the unmarshal of result validation

``` go 
{
	name:   "Test POST request",
	method: http.MethodPost,
	url:    "https://jsonplaceholder.typicode.com/todos/",
	body: ` "Id": 201, "UserId": 2, "Title": "foo", "Body": "bar", "Completed": true`,
	want: ResponseResult{
		statusCode: http.StatusCreated,
		responseBody:  ` "Id": 201, "UserId": 2, "Title": "foo", "Body": "bar", "Completed": true`
	},
},
```

Reply:
BTW you did a great job, thanks! This is just a suggestion. What do you think about it?